### ISSUE TITLE
CSV Importer breaks duplicate detection for new accounts

### ISSUE THREAD

**Issue by mattijs79** [firefly-iii/firefly-iii#3421]
**Bug description**
I am running Firefly III version 5.2.6 and CSV Importer version 1.0.9, and my problem is: when you import a transaction for a new expense account twice it is not recognized as a duplicate

The first time the import is run firefly doesn't know the expense account, so it is created. The second time CSV Importer retrieves the account from firefly, sets the destination_id, and clears destination_name, destination_iban and destination_number (master branch app/Services/Import/Task/Accounts.php line 430) Because the duplicate detection in firefly uses a hash, and hashes all data received from CSV importer, the second hash will be different because the second payload received from CSV importer will be different. Hence, firefly doesn't see it as a duplicate

First import request as received by firefly:
```
The hash is: 7a1a72c40127e1ebd3e8b65e29502e0a3c1ba01dbfd6e1888781dcf16dfb21db {"type":"withdrawal","date":"2020-05-26 00:00:00","order":null,"currency_id":1,"currency_code":null,"foreign_currency_id":null,"foreign_currency_code":null,"amount":"31.160000000000","foreign_amount":null,"description":"Bijdrage A-85 Mei","source_id":1,"source_name":null,"source_iban":null,"source_number":null,"source_bic":null,"destination_id":null,"destination_name":"<RECIPIENT>","destination_iban":"NL34ABNAXXXXXXXXXX","destination_number":null,"destination_bic":null,"budget_id":null,"budget_name":null,"category_id":null,"category_name":null,"bill_id":null,"bill_name":null,"piggy_bank_id":null,"piggy_bank_name":null,"reconciled":false,"notes":null,"tags":["Online bankieren"],"internal_reference":null,"external_id":null,"recurrence_id":null,"bunq_payment_id":null,"sepa_cc":null,"sepa_ct_op":null,"sepa_ct_id":null,"sepa_db":null,"sepa_country":null,"sepa_ep":null,"sepa_ci":null,"sepa_batch_id":null,"interest_date":null,"book_date":"2020-05-26 00:00:00","process_date":null,"due_date":null,"payment_date":null,"invoice_date":null} 
```

Second import request as received by firefly
```
The hash is: 6e37dd0b2b90546c379312358cc178e6427829503dd04de2170c3f2eec9611eb {"type":"withdrawal","date":"2020-05-26 00:00:00","order":null,"currency_id":1,"currency_code":null,"foreign_currency_id":null,"foreign_currency_code":null,"amount":"31.160000000000","foreign_amount":null,"description":"Bijdrage A-85 Mei","source_id":1,"source_name":null,"source_iban":null,"source_number":null,"source_bic":null,"destination_id":4,"destination_name":null,"destination_iban":null,"destination_number":null,"destination_bic":null,"budget_id":null,"budget_name":null,"category_id":null,"category_name":null,"bill_id":null,"bill_name":null,"piggy_bank_id":null,"piggy_bank_name":null,"reconciled":false,"notes":null,"tags":["Online bankieren"],"internal_reference":null,"external_id":null,"recurrence_id":null,"bunq_payment_id":null,"sepa_cc":null,"sepa_ct_op":null,"sepa_ct_id":null,"sepa_db":null,"sepa_country":null,"sepa_ep":null,"sepa_ci":null,"sepa_batch_id":null,"interest_date":null,"book_date":"2020-05-26 00:00:00","process_date":null,"due_date":null,"payment_date":null,"invoice_date":null} 
```

**Comment by JC5** [comment]
I can't replicate this. There have been some changes on the develop branch though; it won't return an account when it's an expense account. Could you check it out?

**Comment by mattijs79** [comment]
I ran the same test on the develop branch, and encountered the same problem. 

My csv:
```
"Datum","Naam / Omschrijving","Rekening","Tegenrekening","Code","Af Bij","Bedrag (EUR)","MutatieSoort","Mededelingen"
"20200526","VvE Kratonkade","NL29INGBXXXXXXXXXX","NL34ABNAXXXXXXXXXX","GT","Af","31,16","Online bankieren","Naam: <RECIPIENT> Omschrijving: Bijdrage A-85 Mei IBAN: NL34ABNA0412529017 Datum/Tijd: 26-05-2020 19:35:16 Valutadatum: 26-05-2020"
```

Import configuration:
```
{
    "date": "Ymd",
    "default_account": 1,
    "delimiter": "comma",
    "headers": true,
    "ignore_duplicate_lines": true,
    "ignore_duplicate_transactions": true,
    "rules": true,
    "skip_form": true,
    "add_import_tag": true,
    "specifics": [
        "IngDescription"
    ],
    "roles": [
        "date_transaction",
        "opposing-name",
        "account-iban",
        "opposing-iban",
        "_ignore",
        "ing-debit-credit",
        "amount",
        "tags-comma",
        "description",
        "date_book",
        "_ignore"
    ],
    "do_mapping": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": {
        "1": [],
        "2": [],
        "3": []
    },
    "version": 2
}
```

**Comment by JC5** [comment]
Mmmm. Thanks for sharing, I'll try it. Are you doing this in two separate runs?

**Comment by mattijs79** [comment]
Yes, I'm doing two separate import runs with CSV importer. Importing the same thing twice. It's easiest to reproduce with a clean database

**Comment by JC5** [comment]
Bingo, I see what you mean.

Don't think there's much I can do. Because of the IBAN mapping Firefly III *will* try to link to the correct account directly, which results in this issue.

I'll think about a solution but don't wait up for it.

**Comment by mattijs79** [comment]
I added this
```
$transaction['destination_id'] = null;
```
as a workaround to app/Services/Import/Task/Accounts.php line 434

It's dirty, but it solves my problem

**Comment by mattijs79** [comment]
Fixed in CSV importer 1.0.10
